### DISCUSSION TITLE
SDK Channel Sideloading, Testing, Packaging, and more!

### DISCUSSION THREAD

**Discussion by ferdiworks** [rrirower/DP2SceneGraphWizard#2] [General]
Converting a Direct Publisher channel to an SDK
1. Using the DP2SceneGraphWizard software
2. Updating the channel manifest with the name of your channel and the URL of your content feed.
3. Adding your channel artwork to the images folder.
4. Testing your channel.
5. Using the Developer Dashboard to switch your channel to SDK, upload your channel package, pass certification
testing, and schedule the submission of your channel.

Steps 
Sideloading the template was found to be fairly easy

Using Putty the Shell application (Windows) or terminal (Mac and Linux) was found to be easy specially following this guide
https://developer.roku.com/docs/developer-program/publishing/packaging-channels.md

Next Packaging the signed package was easy and downloading was simple!

This is where I got stuck because I could not get Roku Developer account to upload the Package file whick is a .pkg file

We must upload the PKG file inorder to run Static Analysis! But the .pkg file will not upload on my DP channel converting to SDK!

**Comment by ferdiworks** [comment]
I understand you wrote that The code will NOT currently pass channel certification.  and you plan on continuing to develop the code and thought it was important to get it into the hands of DP channel developers.

**Comment by rrirower** [comment]
What do you mean that the upload would not work?  Did you get an error code?  Uploading a file is a pretty benign process.

**Reply by ferdiworks** [reply]
See Loom video
https://www.loom.com/share/48a5375e63e84c09b7479ca3b772323d?sid=b1cebf69-24a8-49a8-954c-2d066b5d02cf

**Comment by ferdiworks** [comment]
Today I managed to upload the package and run Static Analysis

Static Analysis Results
everity | Definition
-- | --
Error | Channel issue that will block channel from passing certification.
Warning | Channel issue that does not block certification, but may be reported as an Error in the future.
Info | Helpful information about your channel detected during testing.

#	Message	Severity	Category	Certification requirements

1	Manifest is missing a required attribute: "mm_icon_focus_hd". Path: /manifest.	Error	Manifest	[6.4](https://developer.roku.com/docs/developer-program/certification/certification.md#certification-criteria-1)

2	Channels are required to support roInput events. Please ensure your manifest includes the supports_input_launch=1 entry and your channel properly handles roInput events.	Error	DeepLinking	[5.2](https://developer.roku.com/docs/developer-program/certification/certification.md#certification-criteria-1)

3	"AppLaunchComplete" beacon is missing.	Error	Performance	[3.2](https://developer.roku.com/docs/developer-program/certification/certification.md#certification-criteria-1)

4	If your channel has streamed more than an average of 5 million hours per month over the last three months or you have a new channel projected to reach the specified streaming hour threshold shortly after launch, you must participate in Roku’s Continue Watching program. See Continue Watching for more detail. TVOD, live linear, and made-for-kids channels are excluded from this requirement.	Warning	Uncategorized	[4.13](https://developer.roku.com/docs/developer-program/certification/certification.md#certification-criteria-1)

5	Channel version wasn't updated.	Warning	Manifest	[6.1](https://developer.roku.com/docs/developer-program/certification/certification.md#certification-criteria-1)

6	For your channel to pass certification, your application must fire the "AppDialogInitiate" and "AppDialogComplete" beacons if the channel UI displays a login, user selection, EULA, or any other dialog before the home page.	Warning	Performance	[3.2](https://developer.roku.com/docs/developer-program/certification/certification.md#certification-criteria-1)

**Reply by ferdiworks** [reply]
items 1, 2, 3, are errors!

**Reply by rrirower** [reply]
I don't quite understand why you are getting those errors.  

1. The code outputs a manifest with "mm_icon_focus"hd".  Is your image the correct size?
2. The code includes "supports_input_launch" in the manifest.  However, that's the extent of the code so far.
3. I will add the required code in the next release.

**Reply by ferdiworks** [reply]
The manifest file has the line:
m_icon_focus_hd=pkg:/images/icon_focus_hd.png

I think it needs to be changed to:
mm_icon_focus_hd=pkg:/images/icon_focus_hd.png

**Reply by ferdiworks** [reply]
Yes image is correct size 290 X 218.png

**Reply by rrirower** [reply]
Thanks.  Found the error and fixed it.

**Comment by ferdiworks** [comment]
New Static Analysis Test



Severity | Definition
-- | --
Error | Channel issue that will block channel from passing certification.
Warning | Channel issue that does not block certification, but may be reported as an Error in the future.
Info | Helpful information about your channel detected during testing.



# | Message | Severity | Category | Certification requirements
-- | -- | -- | -- | --
1 | Channels are required to support roInput events. Please ensure your manifest includes the supports_input_launch=1 entry and your channel properly handles roInput events. | Error | DeepLinking | 5.2
2 | "AppLaunchComplete" beacon is missing. | Error | Performance | 3.2
3 | If your channel has streamed more than an average of 5 million hours per month over the last three months or you have a new channel projected to reach the specified streaming hour threshold shortly after launch, you must participate in Roku’s Continue Watching program. See Continue Watching for more detail. TVOD, live linear, and made-for-kids channels are excluded from this requirement. | Warning | Uncategorized | 4.13
4 | Found duplicate attribute in manifest: "minor_version". Path: /manifest. Line: 7. | Warning | Manifest |  
5 | Channel version wasn't updated. | Warning | Manifest | 6.1
6 | Found duplicate attribute in manifest: "build_version". Path: /manifest. Line: 8. | Warning | Manifest |  
7 | Found duplicate attribute in manifest: "major_version". Path: /manifest. Line: 6. | Warning | Manifest |  
8 | For your channel to pass certification, your application must fire the "AppDialogInitiate" and "AppDialogComplete" beacons if the channel UI displays a login, user selection, EULA, or any other dialog before the home page. | Warning | Performance | 3.2

	Message	Severity	Category	Certification requirements
1	Channels are required to support roInput events. Please ensure your manifest includes the supports_input_launch=1 entry and your channel properly handles roInput events.	Error	DeepLinking	[5.2](https://developer.roku.com/docs/developer-program/certification/certification.md#certification-criteria-1)

2	"AppLaunchComplete" beacon is missing.	Error	Performance	[3.2](https://developer.roku.com/docs/developer-program/certification/certification.md#certification-criteria-1)

3	If your channel has streamed more than an average of 5 million hours per month over the last three months or you have a new channel projected to reach the specified streaming hour threshold shortly after launch, you must participate in Roku’s Continue Watching program. See Continue Watching for more detail. TVOD, live linear, and made-for-kids channels are excluded from this requirement.	Warning	Uncategorized	[4.13](https://developer.roku.com/docs/developer-program/certification/certification.md#certification-criteria-1)

4	Found duplicate attribute in manifest: "minor_version". Path: /manifest. Line: 7.	Warning	Manifest	
5	Channel version wasn't updated.	Warning	Manifest	[6.1](https://developer.roku.com/docs/developer-program/certification/certification.md#certification-criteria-1)
6	Found duplicate attribute in manifest: "build_version". Path: /manifest. Line: 8.	Warning	Manifest	

7	Found duplicate attribute in manifest: "major_version". Path: /manifest. Line: 6.	Warning	Manifest	

8	For your channel to pass certification, your application must fire the "AppDialogInitiate" and "AppDialogComplete" beacons if the channel UI displays a login, user selection, EULA, or any other dialog before the home page.	Warning	Performance	[3.2](https://developer.roku.com/docs/developer-program/certification/certification.md#certification-criteria-1)

Error for attribute: "mm_icon_focus_hd". = gone!
Error for input_launch=1 in manifest
Error for "AppLaunchComplete" beacon is missing.
